Mathlib/LinearAlgebra/AffineSpace/AffineSubspace/Defs.lean

```diff
@@ -858,6 +866,12 @@ theorem bot_lt_affineSpan : ⊥ < affineSpan k s ↔ s.Nonempty := by
   rw [bot_lt_iff_ne_bot, nonempty_iff_ne_empty]
   exact (affineSpan_eq_bot _).not
 
+@[simp]
+lemma affineSpan_eq_top_iff_nonempty_of_subsingleton [Subsingleton P] :
+    affineSpan k s = ⊤ ↔ s.Nonempty := by
+  rw [← bot_lt_affineSpan k]
+  exact ⟨fun h ↦ h ▸ bot_lt_top, IsSimpleOrder.eq_top_of_lt⟩
```

Comment:
Could you extract this to a lemma about IsSimpleOrder?

Reply:
Extracted `IsSimpleOrder.bot_lt_iff_eq_top`.